src/ui/include/mainwindow.h

```diff
@@ -205,6 +216,42 @@ private slots:
     QString               m_workingDirectory;
     QMap<QSharedPointer<Extensions::Extension>, QMenu*> m_extensionMenus;
 
+    /**
+	 * @brief Set to true to temporarily disallow updating recent docs. This is useful
+	 *        for loading files that shouldn't be remembered (such as cache files).
+	 */
+    bool                m_dontUpdateRecentDocs = false;
+
+	/**
+	 * @brief Saves a session as an XML file
+	 * @param filePath Path to where the XML file should be created.
+	 * @param cacheModifiedFiles If true, dirty tabs will be written into the cache directory.
+	 *        Only use this for the remember-my-tabs feature. Multiple sessions writing to
+	 *        the cache directory will end up in data loss.
+	 * @return Whether the save has been successful.
+	 */
+    bool                saveSession(QString filePath, bool cacheModifiedFiles);
+
+	/**
+	 * @brief Loads a session XML file and restores all its tabs in the current window.
+	 * @param filePath Path to where the XML file is located.
+	 */
+    void                loadSession(QString filePath);
+
+	/**
+	 * @brief Functions specifically to save/restore tabs to/from cache. These utilize
+	 *        the saveSession and loadSession functions but also save all unsaved progress
+	 *        in the cache.
+	 */
+    bool                saveTabsToCache();
+    void                restoreTabsFromCache();
+
+	/**
+	 * @brief Closes all open tabs but asks the user for permission before discarding changes.
+	 * @return Whether all files have been properly closed.
+	 */
+    bool                closeAllTabs();
```

Comment:
It's important here to specify that this methods doesn't *actually* close the tabs. It acts *as-if*, even saving the changes if the user says so, but never actually closing them.